tests/functional/s3/test_sync_command.py

```diff
@@ -353,6 +353,64 @@ def test_with_copy_props(self):
             ]
         )
 
+    def test_upload_with_checksum_algorithm_sha1(self):
```

Comment:
Let's also add copy (`s3s3`) tests here.

Reply:
Good idea thanks, adding one in next revision